src/Content/VerifiedFeedsDetector.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+// SPDX-FileCopyrightText: 2004-2023 Ryan Parman, Sam Sneddon, Ryan McCue
+// SPDX-License-Identifier: BSD-3-Clause
+
+declare(strict_types=1);
+
+namespace SimplePie\Content;
+
+use SimplePie\Content\Type\Sniffer;
+use SimplePie\File;
+use SimplePie\HTTP\Response;
+use SimplePie\Locator;
+use SimplePie\Registry;
+use SimplePie\RegistryAware;
+use SimplePie\SimplePie;
+
+/**
+ * BC helper for feed auto-discovery and type sniffing
+ *
+ *
+ * This class uses
+ * - \SimplePie\Locator and
+ * - \SimplePie\Content\Type\Sniffer
+ *
+ * @internal
+ */
+final class VerifiedFeedsDetector implements Detector, RegistryAware
+{
+    /**
+     * @var Registry $registry
+     */
+    private $registry;
+
+    public function set_registry(Registry $registry)
+    {
+        $this->registry = $registry;
+    }
+
+    /**
+     * Discover possible feed urls from HTML response
+     *
+     * @see https://simplepie.org/wiki/reference/simplepie/set_autodiscovery_level
+     *
+     * Inspired by the Ultra-liberal RSS locator from Mark Pilgrim
+     * @link http://web.archive.org/web/20110607232437/http://diveintomark.org/archives/2002/08/15/ultraliberal_rss_locator
+     *
+     * @param SimplePie::LOCATOR_* $discovery_level
+     *
+     * @return string[] Array of possible feed urls. The urls are not requested or checked for containing a feed
+     */
+    public function discover_possible_feed_urls(Response $response, int $discovery_level = SimplePie::LOCATOR_ALL): array
+    {
+        /** @var Locator */
+        $locator = $this->registry->create(
```

Comment:
Unlike before, this creates a new locator entry for each method call. But it probably is not very costly and we only call the methods few times anyway.